jsonnet/kube-prometheus/addons/pyrra.libsonnet

```diff
@@ -0,0 +1,651 @@
+            spec: {
+              containers: [c],
+              serviceAccountName: pyrra.kubernetesServiceAccount.metadata.name,
+              nodeSelector: { 'kubernetes.io/os': 'linux' },
+            },
+          },
+        },
+      },
+
+    // Most of these should eventually be moved to the components themselves.
+    // For now, this is a good start to have everything in one place.
+    'slo-apiserver-read-response-errors': {
+      apiVersion: 'pyrra.dev/v1alpha1',
+      kind: 'ServiceLevelObjective',
+      metadata: {
+        name: 'apiserver-read-response-errors',
+        namespace: pyrra._config.namespace,
+        labels: {
+          prometheus: 'k8s',
+          role: 'alert-rules',
+        },
+      },
+      spec: {
+        target: '99',
+        window: '2w',
+        description: '',
+        indicator: {
+          ratio: {
+            errors: {
+              metric: 'apiserver_request_total{job="apiserver",verb=~"LIST|GET",code=~"5.."}',
+            },
+            total: {
+              metric: 'apiserver_request_total{job="apiserver",verb=~"LIST|GET"}',
+            },
+          },
+        },
+      },
+    },
+
+    'slo-apiserver-write-response-errors': {
+      apiVersion: 'pyrra.dev/v1alpha1',
+      kind: 'ServiceLevelObjective',
+      metadata: {
+        name: 'apiserver-write-response-errors',
+        namespace: pyrra._config.namespace,
+        labels: {
+          prometheus: 'k8s',
+          role: 'alert-rules',
+        },
+      },
+      spec: {
+        target: '99',
+        window: '2w',
+        description: '',
+        indicator: {
+          ratio: {
+            errors: {
+              metric: 'apiserver_request_total{job="apiserver",verb=~"POST|PUT|PATCH|DELETE",code=~"5.."}',
+            },
+            total: {
+              metric: 'apiserver_request_total{job="apiserver",verb=~"POST|PUT|PATCH|DELETE"}',
+            },
+          },
+        },
+      },
+    },
+
+    'slo-apiserver-read-resource-latency': {
+      apiVersion: 'pyrra.dev/v1alpha1',
+      kind: 'ServiceLevelObjective',
+      metadata: {
+        name: 'apiserver-read-resource-latency',
+        namespace: pyrra._config.namespace,
+        labels: {
+          prometheus: 'k8s',
+          role: 'alert-rules',
+        },
+      },
+      spec: {
+        target: '99',
+        window: '2w',
+        description: '',
+        indicator: {
+          latency: {
+            success: {
+              metric: 'apiserver_request_duration_seconds_bucket{job="apiserver",scope=~"resource|",verb=~"LIST|GET",le="0.1"}',
+            },
+            total: {
+              metric: 'apiserver_request_duration_seconds_count{job="apiserver",scope=~"resource|",verb=~"LIST|GET"}',
+            },
+          },
+        },
+      },
+    },
+
+    // 'slo-apiserver-read-resource-latency': {
+    //   apiVersion: 'pyrra.dev/v1alpha1',
+    //   kind: 'ServiceLevelObjective',
```

Comment:
Letting this snippet in comments was intended? 🤔

Reply:
Good catch. It conflicted with the name. I think it was a copy-paste mistake. I need to fix it. :+1: